controllers/suite_test.go

```diff
@@ -77,13 +79,18 @@ const (
 
 type K8sClientWrapper struct {
 	client.Client
-	Reader                client.Reader
-	ShouldSimulateFailure bool
+	Reader                  client.Reader
+	ShouldSimulateFailure   bool
+	ShouldSimulateVaFailure bool
 }
 
 func (kcw *K8sClientWrapper) List(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error {
 	if kcw.ShouldSimulateFailure {
 		return errors.New("simulation of client error")
+	} else if kcw.ShouldSimulateVaFailure {
+		if reflect.TypeOf(list) == reflect.TypeOf(&storagev1.VolumeAttachmentList{}) {
```

Comment:
just curious, is there an advantage of using reflect instead of trying a cast?
`if _, ok := list.(*storagev1.VolumeAttachmentList); ok {...}`